### PR TITLE
allow atomic-reactor-repo directory in .dockerignore file,

### PR DESCRIPTION
**Pull request by rcerven**
when it exists

* OSBS-6920

Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: atomic_reactor/util.py [hunk 1]
```diff
@@ -1291,6 +1292,15 @@ def read_yaml(yaml_data, schema):
     return data
 
 
+def allow_repo_dir_in_dockerignore(build_path):
+    docker_ignore = os.path.join(str(build_path), DOCKERIGNORE)
+
+    if os.path.isfile(docker_ignore):
+        with open(docker_ignore, "a") as f:
+            f.write("!%s\n" % RELATIVE_REPOS_PATH)
```
Comment (twaugh) [inline atomic_reactor/util.py]: What if .dockerignore does not already end in a newline character? I think we may want to add one ourselves before the `!`.

File: tests/test_util.py [hunk 2]
```diff
@@ -1510,6 +1511,29 @@ def test_introspection_signal_handler(capfd):
...
+            f.write(ignore_content)
+
+    allow_repo_dir_in_dockerignore(tmpdir)
+
+    if dockerignore_exists:
+        with open(docker_ignore_file, "r") as f:
+            ignore_lines = f.readlines()
+
+        assert len(ignore_lines) == 3
+        assert ignore_lines[2] == added_line
```
Comment (twaugh) [inline tests/test_util.py]: This seems too prescriptive. If all we need is for it to be the last line in the file, and for the existing content to still be in place, we should test those two things:
```
with open(docker_ignore_file, "r") as f:
    # Last line is the one we added.
    assert f.readlines()[-1] == added_line

with open(docker_ignore_file, "r") as f:
    # Previous content is still in place.
    assert f.read(len(ignore_content)) == ignore_content
```

File: atomic_reactor/util.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/util.py]: sure

File: tests/test_util.py [hunk 3]
```diff
@@ -1510,6 +1511,29 @@ def test_introspection_signal_handler(capfd):
     assert log_msg in err
 
 
+@pytest.mark.parametrize('dockerignore_exists', [True, False])
+def test_allow_repo_dir_in_dockerignore(tmpdir, dockerignore_exists):
+    docker_ignore_file = os.path.join(str(tmpdir), DOCKERIGNORE)
+
+    ignore_content = ["# Ignore everything so we can just have a whitelist of things to copy\n",
+                      "** / *\n"]
+    added_lines = ["\n", "!%s\n" % RELATIVE_REPOS_PATH]
```
Comment (twaugh) [inline tests/test_util.py]: Maybe don't include "\n" in `added_lines`. We don't care whether we get an additional newline... we only care that "!..." appears as its own line.

Comment (rcerven) [reply tests/test_util.py]: ok

**Comment by rcerven** [comment]
rn updated : - allow atomic-reactor-repo directory in .dockerignore file, when it exists

**Pull request merged**